binaries/scheduler/main.go

```diff
@@ -1,33 +1,61 @@
 package main
 
 import (
+	"flag"
+	"fmt"
 	"log"
+	"strconv"
+	"strings"
 	"sync"
 
 	"github.com/apache/thrift/lib/go/thrift"
-	//clusterdef "github.com/scootdev/scoot/cloud/cluster"
+	clusterdef "github.com/scootdev/scoot/cloud/cluster"
 	clusterimpl "github.com/scootdev/scoot/cloud/cluster/memory"
 	"github.com/scootdev/scoot/saga"
 	"github.com/scootdev/scoot/sched/distributor"
 	queueimpl "github.com/scootdev/scoot/sched/queue/memory"
 	"github.com/scootdev/scoot/sched/scheduler"
+	"github.com/scootdev/scoot/sched/worker"
 	//"github.com/scootdev/scoot/sched/worker"
 
 	"github.com/scootdev/scoot/sched/worker/fake"
+	"github.com/scootdev/scoot/sched/worker/rpc"
 	// "github.com/scootdev/scoot/sched/worker/rpc"
 	"github.com/scootdev/scoot/scootapi/server"
 )
 
+var addr = flag.String("addr", "localhost:9090", "Bind address for api server.")
+var workers = flag.String("workers", "", "Comma separated list of workers (host:port,...)|NUM:mem.")
+
 func main() {
 	log.Println("Starting Cloud Scoot API Server & Scheduler")
-	addr := "localhost:9090"
+
+	flag.Parse()
+	workersList := strings.Split(*workers, ",")
+	workerNodes := []clusterdef.Node{}
```

Comment:
I like how you let people specify different behaviors in one flag.

This is great for now, so consider what I'm about to say just food for thought for moving forward.

I suspect we'll also want a way to change the cluster implementation (so you can test locally or with static cluster or what not).

I'd make a function like:

func parseCluster(spec string) (cluster.Cluster, workerFactory)

And then this one string will control:
a) static cluster of talking to service discovery
b) what the initial nodes are
c) what worker factory to use

I think that also makes it easy to think about what the language (yes, one string to this is a DSL!) is like.

One thing I've seen work is just using JSON (and golang's easy JSON parsing) to do something like:

{"cluster": {"type": "static"},
 "initial_nodes": ["localhost:2345", "localhost:2346"],
 "workerFactory": "thrift"}

Then there's a parse func that's string -> ClusterSpec, and then another func that's ClusterSpec -> (Cluster, WorkerFactory)

Like I said, none of this is necessary for right now, but worth thinking about how to balance:
a) easy to specify
b) flexible enough to let you specify whatever
c) making the code clear and compartmentalized